source/fe/fe_poly_tensor.cc

```diff
@@ -2140,88 +2168,95 @@ FE_PolyTensor<PolynomialType, dim, spacedim>::requires_update_flags(
 {
   UpdateFlags out = update_default;
 
-  switch (mapping_type)
+  for (unsigned int i = 0; i < this->dofs_per_cell; ++i)
     {
-      case mapping_none:
-        {
-          if (flags & update_values)
-            out |= update_values;
-
-          if (flags & update_gradients)
-            out |= update_gradients | update_values |
-                   update_jacobian_pushed_forward_grads;
-
-          if (flags & update_hessians)
-            out |= update_hessians | update_values | update_gradients |
-                   update_jacobian_pushed_forward_grads |
-                   update_jacobian_pushed_forward_2nd_derivatives;
-          break;
-        }
+      MappingType mapping_type =
```

Comment:
```suggestion
      const MappingType mapping_type =
```

Reply:
I think I got all of them. Lines 395, 975, 1607, 2173 should all have this and also use the convenience function now.